src/v/kafka/server/group.h

```diff
@@ -578,6 +591,7 @@ class group {
     config::configuration& _conf;
     ss::lw_shared_ptr<cluster::partition> _partition;
     absl::node_hash_map<model::topic_partition, offset_metadata> _offsets;
+    absl::node_hash_map<model::topic_partition, group_offset_probe> _probes;
```

Comment:
I think that you should combine these so that the safety involved in keeping a reference in _probes to an offset in _offsets is implicit. You can do this with something like:

```
struct offset_metadata_with_probe {
   offset_metadata offsets;
   group_offset_probe probe;

  offset_metadata_with_probe() : probe(this) {
      // other probe registration.
  }
};

absl::node_hash_map<model::topic_partition, offset_metadata_with_probe> _offsets;
```

Then the probe contains a reference to the object that contains the offset and everything has the exact same lifetime. that also simplifies other stuff like you probably don't have t have calls to things like `create_probe` because the probe is created when the offset is created automatically.